compilation issues after latest pull

Pulled latest version out of git. Can't compile my project as I get some previously non-existing compilation errors. Is something broken here? Or significantly changed. 
I suspect it may have to do something with supporting C++17. When compiling under different rules for CUDA (without C++17) these issues do not arise.

```
In file included from ~/kokkos/core/src/Kokkos_Parallel.hpp:53:0,
~/kokkos/core/src/Kokkos_View.hpp:478:23: error: ‘Kokkos::is_always_assignable_v’ declared as an ‘inline’ variable
 inline constexpr bool is_always_assignable_v =
```


```
In file included from ~/kokkos/core/src/Kokkos_Core.hpp:58:0,
~/kokkos/core/src/Kokkos_OpenMP.hpp:231:8: error: ‘DeviceTypeTraits’ is not a class template
 struct DeviceTypeTraits<OpenMP> {
 ~/kokkos/core/src/Kokkos_OpenMP.hpp:231:33: error: explicit specialization of non-template ‘Kokkos::Profiling::Experimental::DeviceTypeTraits’
 struct DeviceTypeTraits<OpenMP> {
                                 ^
~kokkos/core/src/Kokkos_OpenMP.hpp:232:20: error: ‘DeviceType’ does not name a type
   static constexpr DeviceType id = DeviceType::OpenMP;
```

https://github.com/kokkos/kokkos/blob/874be6a26ed1401245e263b08f26d295e77752a9/cmake/kokkos_corner_cases.cmake#L50-L53
```
$ cmake .. -DCMAKE_CXX_STANDARD=17
-- Setting default Kokkos CXX standard to 17
-- The CXX compiler identification is GNU 6.5.0
-- Check for working CXX compiler: /usr/local/bin/c++
-- Check for working CXX compiler: /usr/local/bin/c++ -- works
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- The project name is: Kokkos
-- SERIAL backend is being turned on to ensure there is at least one Host space. To change this, you must enable another host execution space and configure with -DKokkos_ENABLE_SERIAL=OFF or change CMakeCache.txt
-- Using -std=gnu++1z for C++17 extensions as feature
-- Execution Spaces:
--     Device Parallel: NONE
--     Host Parallel: NONE
--       Host Serial: SERIAL
--
-- Architectures:
-- Found TPLLIBDL: /usr/lib/x86_64-linux-gnu/libdl.so
CMake Error at cmake/kokkos_corner_cases.cmake:52 (MESSAGE):
  You have requested c++17 support for GCC 6.5.0.  Although CMake has allowed
  this and GCC accepts -std=c++1z/c++17, GCC <= 6 does not properly support
  *this capture.  Please reduce the C++ standard to 14 or upgrade the
  compiler if you do need 17 support
Call Stack (most recent call first):
  cmake/kokkos_tribits.cmake:186 (INCLUDE)
  CMakeLists.txt:156 (KOKKOS_SETUP_BUILD_ENVIRONMENT)


-- Configuring incomplete, errors occurred!
See also "/scratch/build_gcc6/CMakeFiles/CMakeOutput.log".
```

> 2nd error reported is related to #3001 and fixed in #3004

I gather this means that this has been solved in release 3.1.1? 
I've updated, but the error persists. 
```
~/kokkos/core/src/Kokkos_OpenMP.hpp:231:8: error: ‘DeviceTypeTraits’ is not a class template
 struct DeviceTypeTraits<OpenMP> {
        ^~~~~~~~~~~~~~~~
~/kokkos/core/src/Kokkos_OpenMP.hpp:231:33: error: explicit specialization of non-template ‘Kokkos::Profiling::Experimental::DeviceTypeTraits’
 struct DeviceTypeTraits<OpenMP> {
                                 ^
~/kokkos/core/src/Kokkos_OpenMP.hpp:232:20: error: ‘DeviceType’ does not name a type; did you mean ‘Device’?
   static constexpr DeviceType id = DeviceType::OpenMP;
                    ^~~~~~~~~~
                    Device
```

Am I missing something?

--- a/core/src/Kokkos_OpenMP.hpp
+++ b/core/src/Kokkos_OpenMP.hpp
@@ -63,6 +63,7 @@
 #include <Kokkos_TaskScheduler.hpp>
 #include <Kokkos_Layout.hpp>
 #include <impl/Kokkos_Tags.hpp>
+#include <impl/Kokkos_Profiling_Interface.hpp>
 
 #include <vector>
 

